blocks/bricks/parallel.py

```diff
@@ -82,10 +82,23 @@ def _push_allocation_config(self):
 
     @application
     def apply(self, *args, **kwargs):
-        args = args + tuple(kwargs[name] for name in
-                            self.input_names[len(args):])
-        return [child.apply(arg)
-                for arg, child in equizip(args, self.children)]
+        # Use of zip() rather than equizip() intentional here.
+        routed_args = dict(zip(self.input_names, args))
```

Comment:
I think that  this logic should be factored out and moved to `/blocks/utils`. E.g.

```python
def extract_args(self, expected_args, args, kwargs):
   """Makes some keyword arguments positional. 

   A frequent situation is that a method of the class gets to 
 
```